http-clients/src/main/java/com/palantir/remoting/http/errors/SerializableErrorErrorDecoder.java

```diff
@@ -112,7 +125,15 @@ private static Exception constructException(Class<? extends Exception> exception
     }
 
     private static Exception constructException(
-            Class<? extends Exception> exceptionClass, String message, Throwable wrappedException) {
+            Class<? extends Exception> exceptionClass, String message, Throwable wrappedException, int status) {
+        if (exceptionClass.equals(ClientErrorException.class)) {
+            return new ClientErrorException(status, wrappedException);
+        } else if (exceptionClass.equals(ServerErrorException.class)) {
+            return new ServerErrorException(status, wrappedException);
+        } else if (exceptionClass.equals(WebApplicationException.class)) {
+            return new WebApplicationException(wrappedException, status);
+        }
```

Comment:
I'm not enthused about having this similar block twice, but since in one case we wrap the exception, more DRY approaches seem much less clean.

Similarly, I'm not a fan of having two different ``if``s for Client and Server, but that lets us remove the reflection and take this out of the try-catch block.